stable/ark-server/templates/deployment.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  namespace: {{ .Values.namespace.name }}
+  name: {{ template "ark-server.fullname" . }}
+  release: {{ .Release.Name }}
+  app: {{ template "ark-server.name" . }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        component: ark
+        release: {{ .Release.Name }}
+        app: {{ template "ark-server.name" . }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+    {{- end }}
+    spec:
+      restartPolicy: Always
+      serviceAccountName: {{ template "ark-server.sa" . }}
+      containers:
+        - name: ark
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          command:
+            - /ark
+          args:
+            - server
+          {{- if eq .Values.configuration.cloudprovider "azure" }}
+          envFrom:
+            - secretRef:
+                name: {{ .Values.secret.name }}
+          {{- end }}
+          volumeMounts:
+            - name: plugins
+              mountPath: /plugins
+          {{- if (or (eq .Values.configuration.cloudprovider "aws") (eq .Values.configuration.cloudprovider "gcp")) }}
+            - name: cloud-credentials
+              mountPath: /credentials
+          env:
+            - name: {{ .Values.configuration.credentials }}
+              value: /credentials/cloud
+          {{- end }}
+      volumes:
+        {{- if (or (eq .Values.configuration.cloudprovider "aws") (eq .Values.configuration.cloudprovider "gcp")) }}
+        - name: cloud-credentials
+          secret:
+            secretName: {{ .Values.secret.name }}
+        {{- end }}
+        - name: plugins
+          emptyDir: {}
+      {{- if eq .Values.configuration.cloudprovider "azure" }}
```

Comment:
I know, but I would like the chart to follow the ark deploy and they use the `nodeselector` for azure only

Reply:
I agree with you since you have managed to squeeze the 3 different provider-specific deployments of Ark into one.
IMHO, you should provide a way to customize the `nodeSelector` while keeping azure's check (and tolerations too as @unguiculus suggests).